### PR TITLE
Check that repositories can only be migrated to own user or organizations

### PR DESCRIPTION
**Pull request by lafriks**
As title

### CODE REVIEW

File: integrations/api_repo_test.go [hunk 1]
```diff
@@ -235,3 +235,28 @@ func TestAPIGetRepoByIDUnauthorized(t *testing.T) {
...
+
+func TestAPIRepoMigrate(t *testing.T) {
+	testCases := []struct {
+		ctxUserID, userID  int64
+		cloneURL, repoName string
+		expectedStatus     int
+	}{
+		{ctxUserID: 2, userID: 2, cloneURL: "https://github.com/go-gitea/git.git", repoName: "git", expectedStatus: http.StatusCreated},
+		{ctxUserID: 2, userID: 1, cloneURL: "https://github.com/go-gitea/git.git", repoName: "git-bad", expectedStatus: http.StatusForbidden},
+		{ctxUserID: 2, userID: 3, cloneURL: "https://github.com/go-gitea/git.git", repoName: "git-org", expectedStatus: http.StatusCreated},
```
Comment (ghost) [inline integrations/api_repo_test.go]: Can you add case when user tries to migrate to org that isn't one that user belongs to?

File: routers/api/v1/repo/repo.go [hunk 2]
```diff
@@ -306,6 +306,11 @@ func Migrate(ctx *context.APIContext, form auth.MigrateRepoForm) {
 		return
 	}
 
+	if !ctxUser.IsOrganization() && ctx.User.ID != ctxUser.ID {
```
Comment (bkcsoft) [inline routers/api/v1/repo/repo.go]: This will disallow an administrator to migrate all users from site A to site B :neutral_face:

Comment (bkcsoft) [reply routers/api/v1/repo/repo.go]: I would wrap this `if`-statement and the next one inside an `if !ctx.User.IsAdmin`-block to fix that

File: integrations/api_repo_test.go (hunk 1 above)
Comment (lafriks) [reply integrations/api_repo_test.go]: done

File: routers/api/v1/repo/repo.go (hunk 2 above)
Comment (lafriks) [reply routers/api/v1/repo/repo.go]: @bkcsoft done

File: integrations/api_repo_test.go (hunk 1 above)
Comment (lafriks) [reply integrations/api_repo_test.go]: @flufmonster done

**Review by bkcsoft — APPROVED** [review]

**Review by axifive — APPROVED** [review]

**Pull request merged**